plenum/common/types.py

```diff
@@ -472,18 +457,30 @@ class ConsistencyProof(MessageBase):
     f.CONS_PROOF
 ])
 
-# class ConsProofRequest(MessageBase):
-#     typename = CONS_PROOF_REQUEST
-#     schema = (
-#         (f.LEDGER_ID.nm, LedgerIdField()),
-#         (f.SEQ_NO_START.nm, NonNegativeNumberField()),
-#         (f.SEQ_NO_END.nm, NonNegativeNumberField()),
-#     )
-ConsProofRequest = TaggedTuple(CONS_PROOF_REQUEST, [
-    f.LEDGER_ID,
-    f.SEQ_NO_START,
-    f.SEQ_NO_END
-])
+
+class MessageReq(MessageBase):
+    """
+    Purpose: ask node for any message
+    """
+    typename = MESSAGE_REQUEST
+    schema = (
+        (f.MSG_TYPE.nm, ChooseField(values={LEDGER_STATUS,
+                                            CONSISTENCY_PROOF, PREPREPARE})),
+        (f.PARAMS.nm, HetroMapField())
+    )
+
+
+class MessageRep(MessageBase):
+    """
+    Purpose: respond to a node for any requested message
+    """
+    typename = MESSAGE_RESPONSE
+    schema = (
+        (f.MSG_TYPE.nm, ChooseField(values={LEDGER_STATUS,
+                                            CONSISTENCY_PROOF, PREPREPARE})),
+        (f.PARAMS.nm, HetroMapField()),
+        (f.MSG.nm, AnyField())
```

Comment:
`AnyField` is used because currently we use it for requesting one message but later it can be used to request several messages, like if had signed Prepares, a node can ask other node to give all Prepares it received (including signature)